deb/build/debian/jenkins.init

```diff
@@ -13,91 +13,54 @@
-	exit 1
-fi
 
-# Make sure there exists a java executable, it may not be always the case
-if [ -z "${JAVA}" ]; then
-	echo "ERROR: No Java executable found in current PATH: ${PATH}" >&2
-	echo "If you actually have java installed on the system make sure the executable is in the aforementioned path and that 'type -p java' returns the java executable path" >&2
-	exit 1
-fi
-
-# Which Java versions can be used to run Jenkins
-JAVA_ALLOWED_VERSIONS=("1.8" "11")
-# Work out the JAVA version we are working with:
-JAVA_VERSION=$(${JAVA} -version 2>&1 | sed -n ';s/.* version "\([0-9]\{2,\}\|[0-9]\.[0-9]\)\..*".*/\1/p;')
-
-if [[ ${JAVA_ALLOWED_VERSIONS[*]} =~ ${JAVA_VERSION} ]]; then
-	echo "Correct java version found" >&2
-else
-	echo "Found an incorrect Java version" >&2
-	echo "Java version found:" >&2
-	${JAVA} -version >&2
-	echo "Aborting" >&2
-	exit 1
-fi
-
-# load environments
-if [ -r /etc/default/locale ]; then
-	. /etc/default/locale
-	export LANG LANGUAGE
-elif [ -r /etc/environment ]; then
-	. /etc/environment
-	export LANG LANGUAGE
-fi
+check_arguments() {
+	# Exit if not supposed to run standalone
+	if [ "${RUN_STANDALONE}" = "false" ]; then
+		echo "ERROR: Not configured to run standalone" >&2
+		return 2
+	fi
 
-# Load the VERBOSE setting and other rcS variables
-. /lib/init/vars.sh
+	# Make sure there exists a java executable, it may not be always the case
+	JAVA=$(command -v "${COMMAND_NAME}")
+	if [ -z "${JAVA}" ]; then
+		echo "ERROR: No Java executable found in current PATH: ${PATH}" >&2
+		echo "If you actually have Java installed on the system make sure the executable is in the aforementioned path and that 'command -v ${COMMAND_NAME}' returns the Java executable path" >&2
+		return 2
+	fi
 
-# Define LSB log_* functions.
-# Depend on lsb-base (>= 3.0-6) to ensure that this file is present.
-. /lib/lsb/init-functions
+	# Work out the JAVA version we are working with:
+	JAVA_VERSION=$("${JAVA}" -version 2>&1 | sed -n ';s/.* version "\([0-9]\{2,\}\|[0-9]\.[0-9]\)\..*".*/\1/p;')
+	if [ "${JAVA_VERSION}" = "1.8" ]; then
+		echo "Correct java version found" >&2
+	elif [ "${JAVA_VERSION}" = "11" ]; then
+		echo "Correct java version found" >&2
+	else
+		echo "Found an incorrect Java version" >&2
+		echo "Java version found:" >&2
+		"${JAVA}" -version >&2
+		echo "Aborting" >&2
+		return 2
+	fi
 
-# Make sure we run as root, since setting the max open files through
-# ulimit requires root access
-if [ "$(id -u)" -gt 0 ]; then
-	echo "The ${NAME} init script can only be run as root"
-	exit 1
-fi
+	# Make sure we run as root, since setting the max open files through
+	# ulimit requires root access
+	if [ "$(id -u)" -gt 0 ]; then
+		echo "The ${NAME} init script can only be run as root" >&2
+		return 2
+	fi
+}
 
 check_tcp_port() {
-	local service=$1
-	local assigned=$2
-	local default=$3
-	local assigned_address=$4
-	local default_address=$5
-	local port
+	service=$1
+	assigned=$2
+	default=$3
+	assigned_address=$4
+	default_address=$5
```

Comment:
`local` is `bash`-specific. We don't depend on the `bash` package, so we shouldn't be using `bash`isms.